src/main/java/com/minecolonies/coremod/client/gui/WindowHutGuardTower.java

```diff
@@ -9,257 +10,448 @@
+    {
+        return "com.minecolonies.coremod.gui.workerHuts.GuardTower";
+    }
+
     /**
-     * Retrieve positions from the building to display in GUI.
+     * Re-sorts the WorkOrders list according to the priorities inside the list.
      */
+    private void sortMobsToAttack()
+    {
+        mobsToAttack.sort(Comparator.comparing(MobEntryView::getPriority, Comparator.reverseOrder()));
+    }
 
+    /**
+     * Retrieve all attributes from the building to display in GUI.
+     */
     private void pullInfoFromHut()
     {
-        this.assignManually = building.assignManually;
-        this.patrolManually = building.patrolManually;
-        this.retrieveOnLowHealth = building.retrieveOnLowHealth;
-        this.task = building.task;
-        this.job = building.job;
-        this.patrolTargets = building.patrolTargets;
+        this.assignManually = building.isAssignManually();
+        this.patrolManually = building.isPatrolManually();
+        this.retrieveOnLowHealth = building.isRetrieveOnLowHealth();
+        this.task = building.getTask();
+        this.job = building.getJob();
+        this.patrolTargets = building.getPatrolTargets();
+        this.mobsToAttack = building.getMobsToAttack();
     }
 
     /**
-     * Handle the task buttons correctly.
+     * Sends the changes to the server.
      */
-    private void handleButtons()
+    private void sendChangesToServer()
     {
-        final Button buttonJob = this.findPaneOfTypeByID(BUTTON_JOB, Button.class);
+        final int ordinal = building.getJob() == null ? -1 : job.ordinal();
+        MineColonies.getNetwork().sendToServer(new GuardTaskMessage(building, ordinal, assignManually, patrolManually, retrieveOnLowHealth, task.ordinal()));
+    }
 
-        if (job != null)
+    /**
+     * Switches whether or not to attack a mob.
+     *
+     * @param button The Switch button clicked
+     */
+    private void switchAttackMode(@NotNull final Button button)
+    {
+        final Label idLabel = (Label) button.getParent().getChildren().get(GUI_LIST_ELEMENT_NAME_POS);
+
+        if (idLabel != null)
         {
-            if (job.equals(AbstractBuildingGuards.GuardJob.KNIGHT))
+            for (final MobEntryView entry : mobsToAttack)
             {
-                buttonJob.setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.knight"));
+                if (entry.getName().equals(idLabel.getLabelText()))
+                {
+                    entry.setAttack(!entry.hasAttack());
+                }
             }
-            else
+            MineColonies.getNetwork().sendToServer(new MobEntryChangeMessage(building, this.mobsToAttack));
+            window.findPaneOfTypeByID(GUI_ELEMENT_LIST_MOBS, ScrollingList.class).refreshElementPanes();
+        }
+    }
+
+    /**
+     * On Button click update the priority.
+     *
+     * @param button the clicked button.
+     */
+    private void updatePriority(@NotNull final Button button)
+    {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 27 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)